reqs_dir.go

```diff
@@ -95,7 +95,11 @@ const readdirResponseHeaderStart = unsafe.Offsetof(ReaddirResponse{}.outHeader)
 const readdirResponseSize = unsafe.Sizeof(ReaddirResponse{})
 
 func readdirResponse(a *Allocator, dataSize uint32) *ReaddirResponse {
-	return (*ReaddirResponse)(a.allocPointer(readdirResponseBaseSize + uintptr(dataSize)))
+	bytes := a.alloc(readdirResponseBaseSize+uintptr(dataSize), false)
```

Comment:
I'll move the commit comment to the code, but basically zeroing is expensive and since calling code may know that it will immediately overwrite a region of bytes, we can choose not to zero. Other times, you're casting to a struct and you want members to be initialized to default/zero values.